presto-orc/src/main/java/com/facebook/presto/orc/reader/LongDictionaryProvider.java

```diff
@@ -75,7 +92,23 @@ public DictionaryResult getDictionary(StreamDescriptor streamDescriptor, long[]
             throws IOException
     {
         InputStreamSource<LongInputStream> dictionaryDataStream = dictionaryStreamSources.getInputStreamSource(streamDescriptor, DICTIONARY_DATA, LongInputStream.class);
-        return loadDictionary(streamDescriptor, requireNonNull(dictionaryDataStream), dictionary, items);
+
+        if (streamDescriptor.getSequence() == 0 || requireNonNull(dictionaryDataStream).openStream() != null) {
+            return requireNonNull(loadDictionary(streamDescriptor, dictionaryDataStream, dictionary, items));
+        }
+
+        // Try fetching shared dictionaries
+        DictionaryEntry entry = sharedDictionaries.get(streamDescriptor.getStreamId());
```

Comment:
See if you can use `sharedDictionaries.computeIfAbsent(...)` instead of `isNew`

Reply:
Do you mean setting the boolean with the input functions to computeIfAbsent? The goal here really is to know whether insertion happens, just for the sake of the return value.

Reply:
Yeah I think the function given to computeIfAbsent is a remapping function. The main goal of isNew here is to record whether an insertion happened or not. Maybe the way I'm writing it is not very idiomatic and I should just use containsKey and live with the additional lookup?

Reply:
computeIfAbsent is only invoked when the key is absent, imho it should work nice in this case. I'm leaving the final decision to use it or not up to you, it's more like a personal preference.